### ISSUE TITLE
Windows build 18363 with WSL2 fails to connect

### ISSUE THREAD

**Issue by piejanssens** [microsoft/vscode-remote-release#6066]
Avid VS Code user here, been running for quite a while without any issues. 
Yesterday this issue suddenly started to appear. 
I have looked at some similar issues, but couldn't find a solution.

```
PS C:\Users\pietejanssens> systeminfo
OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.18363 N/A Build 18363
❯ code --version
1.63.1
fe719cd3e5825bf14e14182fddeb88ee8daf044f
x64
❯ uname -r
5.10.43.3-microsoft-standard-WSL2
❯ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal
PS C:\Users\pietejanssens> wsl -l -v
  NAME                   STATE           VERSION
* Ubuntu                 Running         2
  docker-desktop-data    Stopped         2
  docker-desktop         Stopped         2
```

What I tried so far:
- restart code.exe
- `wsl --shutdown`
- restart PC
- `code --disable-extensions`
- rename ~/.vscode-server (to trigger reconfig/install)

It appears that it's detecting wsl1 instead of wsl2?

WSL debug info
```
2021-12-15 08:56:25.673] Resolving wsl+ubuntu, resolveAttempt: 1
[2021-12-15 08:56:25.900] Starting VS Code Server inside WSL (wsl1)
[2021-12-15 08:56:25.900] Extension version: 0.63.11
[2021-12-15 08:56:25.900] Windows build: 18363. Multi distro support: available. WSL path support: enabled
[2021-12-15 08:56:25.901] No shell environment set or found for current distro.
[2021-12-15 08:56:27.210] WSL daemon log file: c:\Users\pietejanssens\AppData\Roaming\Code\logs\20211215T095619\exthost1\ms-vscode-remote.remote-wsl\wsl-daemon-20211215T085627.log
[2021-12-15 08:56:27.212] Probing if server is already installed: C:\Windows\System32\wsl.exe -d Ubuntu -e sh -c "[ -d ~/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
[2021-12-15 08:56:27.630] Probing result: found
[2021-12-15 08:56:27.630] Server install found in WSL
[2021-12-15 08:56:27.631] Launching C:\Windows\System32\wsl.exe -d Ubuntu sh -c 'env && VSCODE_WSL_DEBUG_INFO=true "$VSCODE_WSL_EXT_LOCATION/scripts/wslServer.sh" fe719cd3e5825bf14e14182fddeb88ee8daf044f stable .vscode-server 680970362672 --host=127.0.0.1 --port=0 --use-host-proxy --without-browser-env-var --disable-websocket-compression --log=trace', cwd: undefined, env: {
[2021-12-15 08:56:27.631]   "ALLUSERSPROFILE": "C:\\ProgramData",
[2021-12-15 08:56:27.631]   "APPDATA": "C:\\Users\\pietejanssens\\AppData\\Roaming",
[2021-12-15 08:56:27.631]   "CHROME_CRASHPAD_PIPE_NAME": "\\\\.\\pipe\\crashpad_21300_ENNTZBCBMYSPWMWW",
[2021-12-15 08:56:27.631]   "CommonProgramFiles": "C:\\Program Files\\Common Files",
[2021-12-15 08:56:27.631]   "CommonProgramFiles(x86)": "C:\\Program Files (x86)\\Common Files",
[2021-12-15 08:56:27.631]   "CommonProgramW6432": "C:\\Program Files\\Common Files",
[2021-12-15 08:56:27.631]   "COMPUTERNAME": "xxx",
[2021-12-15 08:56:27.631]   "ComSpec": "C:\\Windows\\system32\\cmd.exe",
[2021-12-15 08:56:27.631]   "DriverData": "C:\\Windows\\System32\\Drivers\\DriverData",
[2021-12-15 08:56:27.631]   "FPS_BROWSER_APP_PROFILE_STRING": "Internet Explorer",
[2021-12-15 08:56:27.631]   "FPS_BROWSER_USER_PROFILE_STRING": "Default",
[2021-12-15 08:56:27.631]   "HOMEDRIVE": "C:",
[2021-12-15 08:56:27.631]   "HOMEPATH": "\\Users\\pietejanssens",
[2021-12-15 08:56:27.631]   "LOCALAPPDATA": "C:\\Users\\pietejanssens\\AppData\\Local",
[2021-12-15 08:56:27.631]   "LOGONSERVER": "\\\\BEDAC0107",
[2021-12-15 08:56:27.631]   "NUMBER_OF_PROCESSORS": "6",
[2021-12-15 08:56:27.631]   "OneDrive": "C:\\Users\\pietejanssens\\OneDrive - xxx (O365D)",
[2021-12-15 08:56:27.631]   "OneDriveCommercial": "C:\\Users\\pietejanssens\\OneDrive - xxx (O365D)",
[2021-12-15 08:56:27.631]   "ORIGINAL_XDG_CURRENT_DESKTOP": "undefined",
[2021-12-15 08:56:27.631]   "OS": "Windows_NT",
[2021-12-15 08:56:27.631]   "Path": "C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;C:\\Windows\\System32\\OpenSSH\\;C:\\Program Files\\Docker\\Docker\\resources\\bin;C:\\ProgramData\\DockerDesktop\\version-bin;C:\\Users\\pietejanssens\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\pietejanssens\\AppData\\Local\\Programs\\Microsoft VS Code\\bin;C:\\Users\\pietejanssens\\AppData\\Local\\Programs\\Git\\cmd",
[2021-12-15 08:56:27.631]   "PATHEXT": ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC",
[2021-12-15 08:56:27.631]   "PROCESSOR_ARCHITECTURE": "AMD64",
[2021-12-15 08:56:27.631]   "PROCESSOR_IDENTIFIER": "Intel64 Family 6 Model 165 Stepping 2, GenuineIntel",
[2021-12-15 08:56:27.631]   "PROCESSOR_LEVEL": "6",
[2021-12-15 08:56:27.631]   "PROCESSOR_REVISION": "a502",
[2021-12-15 08:56:27.631]   "ProgramData": "C:\\ProgramData",
[2021-12-15 08:56:27.631]   "ProgramFiles": "C:\\Program Files",
[2021-12-15 08:56:27.631]   "ProgramFiles(x86)": "C:\\Program Files (x86)",
[2021-12-15 08:56:27.631]   "ProgramW6432": "C:\\Program Files",
[2021-12-15 08:56:27.631]   "PSModulePath": "C:\\Program Files\\WindowsPowerShell\\Modules;C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules;C:\\Program Files (x86)\\Microsoft Azure Information Protection\\Powershell",
[2021-12-15 08:56:27.631]   "PUBLIC": "C:\\Users\\Public",
[2021-12-15 08:56:27.631]   "SESSIONNAME": "Console",
[2021-12-15 08:56:27.631]   "snow_agent": "C:\\Program Files\\Snow Software\\Inventory\\Agent",
[2021-12-15 08:56:27.631]   "SystemDrive": "C:",
[2021-12-15 08:56:27.631]   "SystemRoot": "C:\\Windows",
[2021-12-15 08:56:27.631]   "TEMP": "C:\\Users\\PIETEJ~1\\AppData\\Local\\Temp",
[2021-12-15 08:56:27.631]   "TMP": "C:\\Users\\PIETEJ~1\\AppData\\Local\\Temp",
[2021-12-15 08:56:27.631]   "UATDATA": "C:\\Windows\\CCM\\UATData\\D9F8C395-CAB8-491d-B8AC-179A1FE1BE77",
[2021-12-15 08:56:27.631]   "USERDNSDOMAIN": "BE.xxx.COM",
[2021-12-15 08:56:27.631]   "USERDOMAIN": "BE",
[2021-12-15 08:56:27.631]   "USERDOMAIN_ROAMINGPROFILE": "BE",
[2021-12-15 08:56:27.631]   "USERNAME": "pietejanssens",
[2021-12-15 08:56:27.631]   "USERPROFILE": "C:\\Users\\pietejanssens",
[2021-12-15 08:56:27.631]   "windir": "C:\\Windows",
[2021-12-15 08:56:27.631]   "ZES_ENABLE_SYSMAN": "1",
[2021-12-15 08:56:27.631]   "VSCODE_WSL_EXT_LOCATION": "c:\\Users\\pietejanssens\\.vscode\\extensions\\ms-vscode-remote.remote-wsl-0.63.11",
[2021-12-15 08:56:27.631]   "WSLENV": "VSCODE_WSL_EXT_LOCATION/up"
[2021-12-15 08:56:27.631] }
[2021-12-15 08:56:27.967] USER=pieter
[2021-12-15 08:56:27.967] VSCODE_WSL_EXT_LOCATION=/mnt/c/Users/pietejanssens/.vscode/extensions/ms-vscode-remote.remote-wsl-0.63.11
[2021-12-15 08:56:27.967] SHLVL=0
[2021-12-15 08:56:27.967] HOME=/home/pieter
[2021-12-15 08:56:27.967] OLDPWD=/mnt/c/Users/pietejanssens/AppData/Local/Programs/Microsoft VS Code
[2021-12-15 08:56:27.967] WSL_DISTRO_NAME=Ubuntu
[2021-12-15 08:56:27.967] LOGNAME=pieter
[2021-12-15 08:56:27.967] WSL_INTEROP=/run/WSL/3558_interop
[2021-12-15 08:56:27.967] NAME=xxx
[2021-12-15 08:56:27.967] _=/usr/bin/sh
[2021-12-15 08:56:27.967] TERM=xterm-256color
[2021-12-15 08:56:27.967] PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Windows/system32:/mnt/c/Windows:/mnt/c/Windows/System32/Wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0/:/mnt/c/Windows/System32/OpenSSH/:/mnt/c/Program Files/Docker/Docker/resources/bin:/mnt/c/ProgramData/DockerDesktop/version-bin:/mnt/c/Users/pietejanssens/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/pietejanssens/AppData/Local/Programs/Microsoft VS Code/bin:/mnt/c/Users/pietejanssens/AppData/Local/Programs/Git/cmd
[2021-12-15 08:56:27.967] LANG=C.UTF-8
[2021-12-15 08:56:27.967] SHELL=/usr/bin/zsh
[2021-12-15 08:56:27.967] PWD=/mnt/c/Users/pietejanssens/AppData/Local/Programs/Microsoft VS Code
[2021-12-15 08:56:27.967] HOSTTYPE=x86_64
[2021-12-15 08:56:27.967] WSLENV=VSCODE_WSL_EXT_LOCATION/up
[2021-12-15 08:56:27.967] Setting up server environment: Looking for /home/pieter/.vscode-server/server-env-setup. Not found.
[2021-12-15 08:56:27.967] WSL version: 5.10.43.3-microsoft-standard-WSL2 Ubuntu
[2021-12-15 08:56:27.967] WSL-shell-PID: 3561
[2021-12-15 08:56:27.967] Node executable: /home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/node
[2021-12-15 08:56:27.967] Starting server: /home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/server.sh --connection-secret=/home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/code-server-3561-1639558587.token --host=127.0.0.1 --port=0 --use-host-proxy --without-browser-env-var --disable-websocket-compression --log=trace
[2021-12-15 08:56:27.967] 
[2021-12-15 08:56:27.967] 
[2021-12-15 08:56:27.967] *
[2021-12-15 08:56:27.967] * Visual Studio Code Server
[2021-12-15 08:56:27.967] *
[2021-12-15 08:56:27.967] * Reminder: You may only use this software with Visual Studio family products,
[2021-12-15 08:56:27.967] * as described in the license https://aka.ms/vscode-remote/license
[2021-12-15 08:56:27.967] *
[2021-12-15 08:56:27.967] 
[2021-12-15 08:56:27.967] Extension host agent listening on 38359
[2021-12-15 08:56:27.967] 
[2021-12-15 08:56:27.967] [09:56:27] Extension host agent started.
[2021-12-15 08:56:27.967] [09:56:27] Started scanning user extensions
[2021-12-15 08:56:27.967] [09:56:27] resolveShellEnv(): running (macOS/Linux)
[2021-12-15 08:56:27.967] [09:56:27] getUnixShellEnvironment#runAsNode undefined
[2021-12-15 08:56:27.967] [09:56:27] getUnixShellEnvironment#noAttach undefined
[2021-12-15 08:56:27.967] [09:56:27] getUnixShellEnvironment#env {
[2021-12-15 08:56:27.967]   USER: 'pieter',
[2021-12-15 08:56:27.967]   VSCODE_WSL_EXT_LOCATION: '/mnt/c/Users/pietejanssens/.vscode/extensions/ms-vscode-remote.remote-wsl-0.63.11',
[2021-12-15 08:56:27.967]   SHLVL: '0',
[2021-12-15 08:56:27.967]   HOME: '/home/pieter',
[2021-12-15 08:56:27.967]   OLDPWD: '/mnt/c/Users/pietejanssens/AppData/Local/Programs/Microsoft VS Code',
[2021-12-15 08:56:27.967]   WSL_DISTRO_NAME: 'Ubuntu',
[2021-12-15 08:56:27.967]   LOGNAME: 'pieter',
[2021-12-15 08:56:27.967]   VSCODE_WSL_DEBUG_INFO: 'true',
[2021-12-15 08:56:27.967]   WSL_INTEROP: '/run/WSL/3558_interop',
[2021-12-15 08:56:27.967]   NAME: 'xxx',
[2021-12-15 08:56:27.967]   _: '/usr/bin/sh',
[2021-12-15 08:56:27.967]   TERM: 'xterm-256color',
[2021-12-15 08:56:27.967]   PATH: '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Windows/system32:/mnt/c/Windows:/mnt/c/Windows/System32/Wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0/:/mnt/c/Windows/System32/OpenSSH/:/mnt/c/Program Files/Docker/Docker/resources/bin:/mnt/c/ProgramData/DockerDesktop/version-bin:/mnt/c/Users/pietejanssens/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/pietejanssens/AppData/Local/Programs/Microsoft VS Code/bin:/mnt/c/Users/pietejanssens/AppData/Local/Programs/Git/cmd',
[2021-12-15 08:56:27.967]   VSCODE_AGENT_FOLDER: '/home/pieter/.vscode-server',
[2021-12-15 08:56:27.967]   LANG: 'C.UTF-8',
[2021-12-15 08:56:27.967]   SHELL: '/usr/bin/zsh',
[2021-12-15 08:56:27.967]   PWD: '/mnt/c/Users/pietejanssens/AppData/Local/Programs/Microsoft VS Code',
[2021-12-15 08:56:27.967]   HOSTTYPE: 'x86_64',
[2021-12-15 08:56:27.967]   WSLENV: 'VSCODE_WSL_EXT_LOCATION/up',
[2021-12-15 08:56:27.967]   VSCODE_INJECT_NODE_MODULE_LOOKUP_PATH: '/home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/remote/node_modules',
[2021-12-15 08:56:27.967]   VSCODE_CWD: '/mnt/c/Users/pietejanssens/AppData/Local/Programs/Microsoft VS Code',
[2021-12-15 08:56:27.967]   ELECTRON_RUN_AS_NODE: '1',
[2021-12-15 08:56:27.967]   ELECTRON_NO_ATTACH_CONSOLE: '1'
[2021-12-15 08:56:27.967] }
[2021-12-15 08:56:27.969] WSL resolver response: ::1:38359
[2021-12-15 08:56:27.969] To debug connection issues, open a local browser on http://[::1]:38359/version
[2021-12-15 08:56:27.969] Using 'localhost' to connect to the VS Code server as configured by setting 'remote.WSL1.connectionMethod'
[2021-12-15 08:56:27.969] No extension host environment set
[2021-12-15 08:56:27.975] + [ ! -d /home/pieter/.vscode-server ]
[2021-12-15 08:56:27.975] + CODE_PROFILE=/home/pieter/.vscode-server/server-env-setup
[2021-12-15 08:56:27.975] + printf Setting up server environment: Looking for %s.  /home/pieter/.vscode-server/server-env-setup
[2021-12-15 08:56:27.975] + [ -f /home/pieter/.vscode-server/server-env-setup ]
[2021-12-15 08:56:27.975] + echo Not found.
[2021-12-15 08:56:27.975] + VSCODE_REMOTE_BIN=/home/pieter/.vscode-server/bin
[2021-12-15 08:56:27.975] + uname -r
[2021-12-15 08:56:27.975] + WSL_VERSION=5.10.43.3-microsoft-standard-WSL2
[2021-12-15 08:56:27.975] + echo WSL version: 5.10.43.3-microsoft-standard-WSL2 Ubuntu
[2021-12-15 08:56:27.975] + dirname /mnt/c/Users/pietejanssens/.vscode/extensions/ms-vscode-remote.remote-wsl-0.63.11/scripts/wslServer.sh
[2021-12-15 08:56:27.975] + /mnt/c/Users/pietejanssens/.vscode/extensions/ms-vscode-remote.remote-wsl-0.63.11/scripts/wslDownload.sh fe719cd3e5825bf14e14182fddeb88ee8daf044f stable /home/pieter/.vscode-server/bin
[2021-12-15 08:56:27.975] + [ ! -d /home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f ]
[2021-12-15 08:56:27.975] + RC=0
[2021-12-15 08:56:27.975] + [ 0 -ne 0 ]
[2021-12-15 08:56:27.975] + date +%s
[2021-12-15 08:56:27.975] + TOKEN_FILE_PATH=/home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/code-server-3561-1639558587.token
[2021-12-15 08:56:27.975] + echo 680970362672
[2021-12-15 08:56:27.975] + echo WSL-shell-PID: 3561
[2021-12-15 08:56:27.975] + echo Node executable: /home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/node
[2021-12-15 08:56:27.975] + echo Starting server: /home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/server.sh --connection-secret=/home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/code-server-3561-1639558587.token --host=127.0.0.1 --port=0 --use-host-proxy --without-browser-env-var --disable-websocket-compression --log=trace
[2021-12-15 08:56:27.975] + [ -f /etc/alpine-release ]
[2021-12-15 08:56:27.975] + uname -m
[2021-12-15 08:56:27.975] + [ x86_64 = aarch64 ]
[2021-12-15 08:56:27.975] + export VSCODE_AGENT_FOLDER=/home/pieter/.vscode-server
[2021-12-15 08:56:27.975] + /home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/server.sh --connection-secret=/home/pieter/.vscode-server/bin/fe719cd3e5825bf14e14182fddeb88ee8daf044f/code-server-3561-1639558587.token --host=127.0.0.1 --port=0 --use-host-proxy --without-browser-env-var --disable-websocket-compression --log=trace
```

**Comment by rafalhejna** [comment]
I have the same issue. I've downgraded the version of `Remote - WSL` plugin to `0.63.5` and that helped.

**Comment by piejanssens** [comment]
Hello @rafalhejna,

Thanks, that solves it!

@aeschli It appears there is a regression in 0.63.11

**Comment by danw-mpl** [comment]
Appears related to https://github.com/microsoft/vscode-remote-release/issues/6067

**Comment by piejanssens** [comment]
For me rollback to v0.63.5 was OK.

What'I'm still seeing, now that it works again and thus apparently unrelated
```
Starting VS Code Server inside WSL (**wsl1**)
```

**Comment by deesejohn** [comment]
Same here, downgraded to `0.63.5` solved it. Windows 10 with Debian Testing under wsl2. Can provide debug logs if that would be helpful.

**Comment by aeschli** [comment]
The issue is with not detecting that this is WSL2.
I see you are on Windows build: 18363. I assumed that WSL came in 18917, but I just learned about https://devblogs.microsoft.com/commandline/wsl-2-support-is-coming-to-windows-10-versions-1903-and-1909/

So I'll fix this.

@deesejohn @danw-mpl @rafalhejna Are you also on  Windows build 18363, using a WSL 2 distro?

**Comment by rafalhejna** [comment]
@aeschli Yes, I'm on Windows build 18363 with WSL2.

**Comment by deesejohn** [comment]
@aeschli that appears to be the issue: `OS Version:                10.0.18363 N/A Build 18363`

**Comment by danw-mpl** [comment]
@aeschli Yes, 18363 with WSL 2.

**Comment by aeschli** [comment]
I published Remote WSL 0.63.12. It should fix the issue with build 18363 and WSL2. 
Please give it a try and let me know.

**Comment by piejanssens** [comment]
@aeschli  it's fixed with 0.63.12 ✔

**Comment by deesejohn** [comment]
I successfully updated to `0.63.12` as well, thank you for the quick fix @aeschli 🥇
